laplace/baselaplace.py

```diff
@@ -429,21 +438,20 @@ def _gridsearch(
         results = list()
         prior_precs = list()
         pbar = tqdm.tqdm(interval) if progress_bar else interval
+
         for prior_prec in pbar:
             self.prior_precision = prior_prec
-            try:
```

Comment:
Maybe it's worth keeping the try-except statement for its original purpose, to catch inversion errors for a prior precision that is too small, so that the gridsearch is not interrupted, but the next value is tried. So it could be something like this (this would work when using Cholesky for the inversions):
```python
try:
    ...
except RuntimeError as error:
    if "not positive-definite" in str(error):
        result = np.inf
    else:
        raise error
```